### DISCUSSION TITLE
Personalized link in alert_text

### DISCUSSION THREAD

**Discussion by ngms17** [jertel/elastalert2#1039] [Q&A] [answered]
Currently i configured a personalized URL that opens a filtered Discover view from my kibana interface. Instead of displaying this enormous link on my email alert i want to display this link but with a personalized and smaller text (like for example "Click here to see the alert"). I have tried with a href but i have been not able to configure. This is the error:

ec 05 14:05:56 suricata-elk01 elastalert[26922]: yaml.parser.ParserError: while parsing a block mapping
Dec 05 14:05:56 suricata-elk01 elastalert[26922]:   in "<unicode string>", line 1, column 1:
Dec 05 14:05:56 suricata-elk01 elastalert[26922]:     name: Critical Activity Detected
Dec 05 14:05:56 suricata-elk01 elastalert[26922]:     ^
Dec 05 14:05:56 suricata-elk01 elastalert[26922]: expected <block end>, but found '<scalar>'
Dec 05 14:05:56 suricata-elk01 elastalert[26922]:   in "<unicode string>", line 60, column 43:
Dec 05 14:05:56 suricata-elk01 elastalert[26922]:      ... : "**** ELASTALERT ****<a href="https://suricata

This is the rule config file:

alert_text: "**** ELASTALERT ****\n\nLINK: https://xxxx:5601/app/discover#/view/cfbdd150-6a8d-11ed-8853-470b6a3cfc32?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'{0}',to:now))&_a=(columns:!(suricata.eve.flow_id,suricata.eve.event_type,source.ip,source.port,source.geo.country_name,destination.ip,destination.port,destination.geo.country_name,source.address,destination.address,evebox),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:'suricata-ids-*',key:event.kind,negate:!f,params:!(alert,event),type:phrases),query:(bool:(minimum_should_match:1,should:!((match_phrase:(event.kind:alert)),(match_phrase:(event.kind:event)))))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:'suricata-ids-*',key:suricata.eve.flow_id,negate:!f,params:(query:'803338954570541'),type:phrase),query:(match_phrase:(suricata.eve.flow_id:'{1}')))),grid:(columns:(suricata.eve.flow_id:(width:196))),hideChart:!f,index:'suricata-ids-*',interval:auto,query:(language:kuery,query:''),sort:!(!('@timestamp',desc)))\n\nTimestamp: {0}\n\nFlow ID: {1}\n\nSignature: {2}\n\nSignature_ID: {3}\n\nCategory: {4}\n\nSource Address: {5}:{6} | Country: {7}\n\nDestination Address: {8}:{9} | Country: {10}\n\nNetwork protocol: {11}\n\nNetwork transport: {12}\n\nBytes: {13}\n\nPackets: {14}\n\n"

**Comment by jertel** [comment, marked as answer]
ElastAlert 2 supports Kibana URL shortening natively. See the [docs](https://elastalert2.readthedocs.io/en/latest/ruletypes.html#kibana-url) for more information.

**Reply by ngms17** [reply]
But how can i insert personalized filters to the natively generated URL? Because i want to open the link and show specific information from that alert only

**Reply by jertel** [reply]
If you are trying to embed a custom link into an alert text and it's failing to parse then most likely you have a quotation issue. I cannot easily read the detailed rule configuration in your original post since you didn't follow the guidelines #11. I suggest reviewing your double quotes and single quotes.
